src/btree/bt_cursor.c

```diff
@@ -683,6 +683,14 @@ __wt_btcur_search_near(WT_CURSOR_BTREE *cbt, int *exactp)
         }
     }
 
+    /*
+     * If a valid key has been found and we are doing a prefix search near, we want to return the
+     * key only if it is a prefix match.
+     */
+    if (valid && F_ISSET(cursor, WT_CURSTD_PREFIX_SEARCH) &&
+      __wt_prefix_match(&cbt->iface.key, cbt->tmp) != 0)
```

Comment:
I wonder if we should use `&state.key` here instead of `&cbt->iface.key`. To me, it seems to be safer to use a saved state of the key, although im not sure if cbt->iface.key has the possibility to change. What are your thoughts @etienneptl ?

Reply:
I think you are right, @luke-pearson, is it "safer" to use `&state.key`? I can see it is being used further down in `__wt_btcur_next_prefix` and `__wt_btcur_prev_prefix`.

Reply:
`state.key` is correct here.

Reply:
Thank you both, fixed in 3940747f88217aa2786e9ee5ac1deb65ab0638f3